genienlp/data_manipulation_scripts/dialog_to_tsv.py

```diff
@@ -0,0 +1,92 @@
+from argparse import ArgumentParser
+import csv
+
+
+def read_dialog_file(dialog_file, args):
+    all_prompts = {}
+    dialog_lines = []
+    with open(dialog_file) as f:
+        for l in f:
+            dialog_lines.append(l.strip())
+
+    idx = 0
+    dialog_idx = 0
+    while idx < len(dialog_lines):
+        if dialog_lines[idx] == args.dialog_start:
+            # new dialog started
+            dialog_start_line_idx = idx
+            idx += 1
+            while idx < len(dialog_lines) and dialog_lines[idx] != args.dialog_start:
+                idx += 1
+            prompts = read_dialog(dialog_lines[dialog_start_line_idx: idx], args)
+            all_prompts[dialog_idx] = prompts
+            dialog_idx += 1
+    return all_prompts
+
+def read_dialog(lines: list, args):
+    prompts = []
+    agent_utterances = ['']
+    previous_ats = []
+    next_uts = []
+    user_utterance = None
+    # index = 0
+    for idx, l in enumerate(lines):
+        # if l.startswith(args.index_prefix):
+            # index = l[len(args.index_prefix):].strip()
+        if l.startswith(args.at_prefix):
+            previous_ats.append(l[len(args.at_prefix):].strip())
+        if l.startswith(args.ut_prefix):
+            next_uts.append(l[len(args.ut_prefix):].strip())
+            if idx == len(lines)-1 or not lines[idx+1].startswith(args.ut_prefix):
+                # this is the last UT, so complete the example
+                if len(previous_ats) == 0:
+                    previous_ats.append('null')
+                previous_ats = []
+                next_uts = []
+        if l.startswith(args.agent_prefix):
+            agent_utterances.append(l[len(args.agent_prefix):].strip())
+        if l.startswith(args.user_prefix):
+            user_utterance = l[len(args.user_prefix):].strip()
+            last_agent_utterance = agent_utterances[-1]
+            prompts.append((last_agent_utterance, user_utterance))
+
+    return prompts
```

Comment:
This whole file and this code would be dramatically easier if written in JavaScript, because Genie already has code to parse its own dialogue format, which also appears to be more robust to invalid inputs than this.